opencensus/proto/trace/trace_config.proto

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package opencensus.proto.trace;
+
+option java_multiple_files = true;
+option java_package = "io.opencensus.proto.trace";
+option java_outer_classname = "TraceConfigProto";
+
+option go_package = "github.com/census-instrumentation/opencensus-proto/gen-go/traceconfigproto";
+
+// Global configuration of the trace service.
+message TraceConfig {
+
+  // The global default sampler used to make decisions on span sampling.
+  oneof sampler {
+    ProbabilitySampler probability_sampler = 1;
+
+    AlwaysSampler always_sampler = 2;
```

Comment:
I think something like "ConstantSampler constant_sampler = 2;" can replace always/never and we have ConstantSampler {
  boolean decision = 1;
}

This way we do not have empty sampler messages and I think will look better.

Reply:
Make sense to me.